MVP Module/Source/Feed Presentation/FeedPresenter.swift

```diff
@@ -13,17 +13,25 @@ protocol FeedView {
 	func display(_ viewModel: FeedViewModel)
 }
 
+protocol FeedErrorView {
+	func display(_ viewModel: FeedErrorViewModel)
+	func hide()
+}
```

Comment:
Suggestion:
It can be better to have `FeedErrorViewModel.errorMessage` as optional and hide the view by calling `display(.init(errorMessage: nil))`, because `hide` is a very generic name and can have conflicts with other methods in the future. 
Also it will keep all the state transitions centralized into a single state type.

```suggestion
protocol FeedErrorView {
	func display(_ viewModel: FeedErrorViewModel)
}
```

Reply:
@danillahtin Thanks for your comments!

I was actually thinking about that approach.
But what got me off the track with that solution is that I have to make the message of an optional String type.
And the problem with that is it's not explicit. 

For instance, for me to read this code, and see `hide` and `display` methods are more self-documented and explicit than having a single `display` method that can either hide or display.
Single method `display` that can also hide, in my opinion, is a violation of SoC principle. 

To not cause any overriding issues in the future and conform to proper polymorphism rules, I can rename the `hide` method to `hideErrorView`.

Let me know what you think, and if you still want me to make those changes, I'll do them gladly.

Thanks!

Reply:
Hi @ludoed,
All good, it was just a suggestion, either ways are correct 👍